components/typography/Text.js

```diff
@@ -43,10 +45,12 @@ const factory = (baseType, type, defaultElement) => {
 
       const Element = element || defaultElement;
 
+      const rest = omit(others, ['soft']);
```

Comment:
This seems a bit tedious to check every component to make sure not to send too many props down? We are checking in `<Box>` itself which props are supported, so not sure if this is necessary?